lib/configuration.js

```diff
@@ -15,150 +15,430 @@
- * @property {String} version - the hosting applications version
+ * The Configuration constructor function initializes several internal
+ * properties on the Configuration instance and accepts a runtime-given
+ * configuration object which may be used by the Configuration instance
+ * depending on the initialization transaction that occurs with the meta-data
+ * service.
+ * @class Configuration
+ * @extends EventEmitter
+ * @classdesc The Configuration class represents the runtime configuration of
+ *  the Stackdriver error handling library. This Configuration class accepts the
+ *  configuration options potentially given through the application interface
+ *  but it also preferences values received from the metadata service over
+ *  values given through the application interface. Becuase the Configuration
+ *  class must handle async network I/O it exposes an EventEmitter interface
+ *  wherein components interested in waiting for the instances configuration
+ *  resolution may listen to the `ready` event. The `error` event may be emitted
+ *  place of the `ready` event in which case the Configuration instance has
+ *  failed to gather necessary information to interact with the Stackdriver
+ *  service and all listening components must proceed by operating in `offline`
+ *  mode.
+ * @param {ConfigurationOptions} givenConfig - The config given by the
+ *  hosting application at runtime. Configuration values will only be observed
+ *  if they are given as a plain JS object; all other values will be ignored.
+ * @param {commonDiag.logger} logger - an inited logger instance
  */
-
+var Configuration = function(givenConfig, logger) {
+  /**
+   * The _reportUncaughtExceptions property is meant to contain the optional
+   * runtime configuration property reportUncaughtExceptions. This property will
+   * default to true if not given false through the runtime configuration
+   * meaning that the default behavior is to catch uncaught exceptions, report
+   * them to the Stackdriver Errors API and then exit. If given false uncaught
+   * exceptions will not be listened for and not be caught or reported.
+   * @memberof Configuration
+   * @private
+   * @type {Boolean}
+   * @defaultvalue true
+   */
+  this._reportUncaughtExceptions = true;
+  /**
+   * The _shouldReportErrorsToAPI property is meant to denote whether or not
+   * the Stackdriver error reporting library will actually try to report Errors
+   * to the Stackdriver Error API. The value of this property is derived from
+   * the `NODE_ENV` environmental variable. If the `NODE_ENV` variable is set to
+   * 'production' then the _shouldReportErrorToAPI property will be set to true
+   * and error reporting library will attempt to send errors to the Error API.
+   * Otherwise the value will remain false and errors will not be reported to
+   * the API.
+   * @memberof Configuration
+   * @private
+   * @type {Boolean}
+   */
+  this._shouldReportErrorsToAPI = env.NODE_ENV === 'production';
```

Comment:
I think _enabled is too general. What is enabled? The class? A property of the class? Makes it seem like the class itself is enabled when it actually has nothing to do with the class itself - the class is instead simply a container for another explicit configuration